client/components/tinymce/plugins/wpcom-view/wpvideo-view-container.jsx

```diff
@@ -0,0 +1,88 @@
+/**
+ * External dependencies
+ */
+import React, { Component, PropTypes } from 'react';
+
+/**
+ * Internal dependencies
+ */
+import shortcodeUtils from 'lib/shortcode';
+import MediaActions from 'lib/media/actions';
+import MediaListStore from 'lib/media/list-store';
+import WpVideoView from './wpvideo-view';
+
+const getStateData = siteId => {
+	return {
+		media: MediaListStore.getAll( siteId ),
+		mediaHasNextPage: MediaListStore.hasNextPage( siteId ),
+		mediaFetchingNextPage: MediaListStore.isFetchingNextPage( siteId )
+	};
+};
+
+class WpVideoViewContainer extends Component {
+
+	static match( content ) {
+		const match = shortcodeUtils.next( 'wpvideo', content );
+
+		if ( match ) {
+			return {
+				index: match.index,
+				content: match.content,
+				options: {
+					shortcode: match.shortcode
+				}
+			};
+		}
+	}
+
+	static serialize( content ) {
+		return encodeURIComponent( content );
+	}
+
+	constructor( props ) {
+		super( props );
+		this.state = getStateData( props.siteId );
+	}
+
+	componentDidMount() {
+		MediaListStore.on( 'change', this.updateStateData.bind( this ) );
+		this.fetchData( this.props.siteId );
+	}
+
+	componentWillUnmount() {
+		MediaListStore.off( 'change', this.updateStateData.bind( this ) );
+	}
+
+	componentWillReceiveProps( nextProps ) {
+		if ( this.props.siteId !== nextProps.siteId ) {
+			this.fetchData( this.props.siteId );
+			this.updateStateData();
+		}
+	}
+
+	fetchData( siteId ) {
+		setTimeout( () => {
```

Comment:
Defaulting to a fixed width value would be the easiest. I chose not to do so because shortcodes can have a `w` and `h` (width and height, respectively) attribute that would be ignored in the preview but would take effect in the post preview/published post, so in the end, the in-editor preview would show something different than what we would see in the post preview/published post. It's a trade-off between having a not-so-realistic preview vs. adding code complexity to make the preview resemble the reality closer.